### DISCUSSION TITLE
cache data disappear on page navigation

### DISCUSSION THREAD

**Discussion by paddyfink** [TanStack/query#4954] [Q&A] [answered]
Using this example: https://tanstack.com/query/v4/docs/react/examples/react/star-wars

when we navigate between pages, the cached data are visible in the devtools.

But in my app with pretty much the same configuration, when I navigate away from a page, I can see the cache disappearing in the devtools .

I don't understand why. Does someone have an idea?

the issue is pretty similar to this one: https://www.reddit.com/r/reactjs/comments/uqd6g3/react_query_clearing_cache_when_navigating_away/

**Comment by louis-young** [comment]
Hi 👋 

It's difficult to reason about this without looking at your implementation. Please can you provide a minimal runnable sandbox so we can take a look for you?

**Comment by paddyfink** [comment]
That's the thing, if I create a minimal implementation, it will work. I have the very same configuration as the star wars example.
So there is something else I think, and unfortunately, I cannot provide the whole codebase.

I was looking for broad information about cases that can make the cache disappear so that I can investigate furthur

**Comment by paddyfink** [comment, marked as answer]
I found the problem. It is the same as https://github.com/TanStack/query/issues/2765.

I was creating the client inside the app
